services/ymax-planner/test/kv-store-flow.test.ts

```diff
@@ -0,0 +1,255 @@
+  const destinationAddress = `eip155:42161:${contractAddress}`;
+  const txId = 'tx2' as `tx${number}`;
+
+  const gmpTx = createMockPendingTxData({
+    type: TxType.GMP,
+    destinationAddress,
+  });
+
+  const chainId = 'eip155:42161';
+  const opts = createMockPendingTxOpts();
+  const mockProvider = opts.evmProviders[chainId] as any;
+
+  const currentTimeMs = 1700000000; // 2023-11-14T22:13:20Z
+  const txTimestampMs = currentTimeMs - 10 * 1000; // 10 seconds ago
+  const avgBlockTimeMs = 300; // 300 ms per block on eip155:42161
+
+  const latestBlock = 1_450_131;
+  mockProvider.getBlockNumber = async () => latestBlock;
+
+  mockProvider.getBlock = async (blockNumber: number) => {
+    const blocksAgo = latestBlock - blockNumber;
+    const ts = currentTimeMs - blocksAgo * avgBlockTimeMs;
+    return { timestamp: Math.floor(ts / 1000) };
+  };
+
+  // Trigger block event to resolve waitForBlock
+  setTimeout(() => mockProvider.emit('block', latestBlock + 1), 10);
+
+  const event = createMockGmpExecutionEvent(txId);
+  mockProvider.getLogs = async (args: {
+    fromBlock: number;
+    toBlock: number;
+  }) => {
+    const blockWithEvent = 1_450_080;
+    if (blockWithEvent >= args.fromBlock && blockWithEvent <= args.toBlock)
+      return [event];
+    else return [];
+  };
+
+  const ctxWithFetch = harden({
+    ...opts,
+    fetch: async (url: string) => {
+      return {
+        ok: true,
+        json: async () => ({
+          data: [
+            {
+              status: 'executed',
+              call: {
+                transactionHash: '0xabcdef123456',
+                returnValues: {
+                  messageId: `msg_${txId}`,
+                },
+              },
+              executed: {
+                transactionHash: '0xexecuted123',
+                receipt: {
+                  logs: [createMockGmpExecutionEvent(txId)],
+                },
+              },
+            },
+          ],
+        }),
+      } as Response;
+    },
+  });
+
+  const initialMulticallExecutedSavedBlock = await getResolverLastActiveBlock(
+    ctxWithFetch.kvStore,
+    txId,
+    EVENTS.MULTICALL_EXECUTED,
+  );
+  const initialMulticallStatusSavedBlock = await getResolverLastActiveBlock(
+    ctxWithFetch.kvStore,
+    txId,
+    EVENTS.MULTICALL_STATUS,
+  );
+
+  t.is(initialMulticallExecutedSavedBlock, undefined);
+  t.is(initialMulticallStatusSavedBlock, undefined);
+
+  await handlePendingTx(
+    { txId, ...gmpTx },
+    {
+      ...ctxWithFetch,
+      log: mockLog,
+    },
+    txTimestampMs,
+  );
+
+  const finalMulticallExecutedSavedBlock = await getResolverLastActiveBlock(
+    ctxWithFetch.kvStore,
+    txId,
+    EVENTS.MULTICALL_EXECUTED,
+  );
+  const finalMulticallStatusSavedBlock = await getResolverLastActiveBlock(
+    ctxWithFetch.kvStore,
+    txId,
+    EVENTS.MULTICALL_STATUS,
+  );
+
+  t.is(finalMulticallExecutedSavedBlock, 1450079);
+  t.is(finalMulticallStatusSavedBlock, 1450079);
```

Comment:
Number consistency:
```suggestion
  t.is(finalMulticallExecutedSavedBlock, 1_450_079);
  t.is(finalMulticallStatusSavedBlock, 1_450_079);
```

(but honestly, it would be preferable to just use heights less than 100)